test/e2e/operator/nmstate_install_test.go

```diff
@@ -0,0 +1,158 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/apps/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1beta1"
+	"github.com/nmstate/kubernetes-nmstate/test/e2e/daemonset"
+	"github.com/nmstate/kubernetes-nmstate/test/e2e/deployment"
+)
+
+var _ = Describe("NMState operator", func() {
+	Context("when installed for the first time", func() {
+		BeforeEach(func() {
+			installDefaultNMState()
+		})
+		It("should deploy daemonset and webhook deployment", func() {
+			daemonset.DaemonSet(getDefaultNMStateHandlerKey()).ReadyConditionEventually().Should(BeTrue())
+			deployment.Deployment(getDefaultNMStateWebhookKey()).ReadyConditionEventually().Should(BeTrue())
+		})
+	})
+	Context("when NMState is installed", func() {
+		BeforeEach(func() {
+			ds, err := getDefaultNMStateHandlerList()
+			Expect(err).ToNot(HaveOccurred(), "List daemon sets in namespace nmstate succeeds")
+			Expect(ds.Items).To(HaveLen(1), "and has only one item")
+		})
+		Context("and the CR is created with a wrong name", func() {
+			var nmstate = getNMStateWrongName()
+			BeforeEach(func() {
+				err := framework.Global.Client.Create(context.TODO(), &nmstate, &framework.CleanupOptions{})
+				Expect(err).ToNot(HaveOccurred(), "NMState CR with incorrect name is created without error")
+			})
+			AfterEach(func() {
+				err := framework.Global.Client.Delete(context.TODO(), &nmstate, &client.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred(), "NMState CR with incorrect name is removed without error")
+			})
+			It("should ignore it", func() {
+				ds, err := getDefaultNMStateHandlerList()
+				Expect(err).ToNot(HaveOccurred(), "Daemon set list is retreieved without error")
+				Expect(ds.Items).To(HaveLen(1), "and still only has one item")
+			})
+		})
+	})
+	Context("when NMState is uninstalled", func() {
+		BeforeEach(func() {
+			uninstallDefaultNMState()
+		})
+		It("should uninstall handler and webhook", func() {
+			Eventually(func() bool {
+				_, err := getDefaultNMStateHandler()
+				return apierrors.IsNotFound(err)
+			}, 60*time.Second, 1*time.Second).Should(BeTrue(), fmt.Sprint("Daemon set for NMState should be removed, but is not"))
+			Eventually(func() bool {
+				_, err := getDefaultNMStateWebhook()
+				return apierrors.IsNotFound(err)
+			}, 60*time.Second, 1*time.Second).Should(BeTrue(), fmt.Sprint("Deployment for NMState webhook should be removed, but is not"))
+		})
+	})
+})
```

Comment:
Move this to `AfterEach`of the parent `Context` so we just check there that handler and webhook are gone, I know is better to have `It` but is more convenient to have the AfterEach.